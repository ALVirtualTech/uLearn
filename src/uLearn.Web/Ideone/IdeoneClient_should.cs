using System;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using NUnit.Framework;

namespace uLearn.Web.Ideone
{
	[TestFixture]
	public class IdeoneClient_should
	{
		private readonly IdeoneClient service = new IdeoneClient(TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(30));
		
		[Test, UseReporter(typeof(DiffReporter))]
		[Explicit]
		public void make_submition()
		{
			GetSubmitionDetailsResult res =
				service.Submit(@"using System; public class M{static void Main(){System.Console.WriteLine(42);}}",
					"")
					.Result;
			Approvals.Verify(res.WritePropertiesToString());
		}
		
		[Test]
		[Explicit]
		public void compile_tuple()
		{
			GetSubmitionDetailsResult res =
				service.Submit(@"using System; public class M{static void Main(){System.Console.WriteLine(Tuple.Create(1, 2));}}",
					"")
					.Result;
			Console.WriteLine(res.Error);
			Console.WriteLine(res.Status);
			Console.WriteLine(res.Result);
			Console.WriteLine(res.LangId);
			Console.WriteLine(res.LangName);
			Console.WriteLine(res.LangVersion);
			Console.WriteLine(res.IsPublic);
			Console.WriteLine(res.Date);
			Console.WriteLine(res.Source);
			Console.WriteLine(res.CompilationError);
			Console.WriteLine(res.Input);
			Console.WriteLine(res.Output);
			Console.WriteLine(res.StdErr);
			Console.WriteLine(res.Signal);
			Console.WriteLine(res.Time);
			Console.WriteLine(res.Memory);
			Assert.AreEqual(SubmitionResult.Success, res.Result);
		}

		[Test]
		[Explicit]
		public void get_supported_languages()
		{
			foreach (var lang in service.GetSupportedLanguages())
			{
				Console.WriteLine(lang);
			}
		}

	}

}