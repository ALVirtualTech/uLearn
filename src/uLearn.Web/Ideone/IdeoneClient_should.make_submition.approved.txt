GetSubmitionDetailsResult
{
	LangId: 27
	LangName: C#
	LangVersion: mono-4.0.2
	Time: 00:00:00.0400000
	Memory: 23880
	Signal: 0
	IsPublic: False
	Date: 30.07.2015 10:09:09
	Source: using System; public class M{static void Main(){System.Console.WriteLine(42);}}
	Input: 
	Output: 42

	StdErr: 
	CompilationError: 
	Status: Done
	Result: Success
	IsOk: True
	Error: OK
}
